### DISCUSSION TITLE
Unable to read key from file file:///tmp/storage/oauth-private.key

### DISCUSSION THREAD

**Discussion by Gurdarshankonspot** [brefphp/bref#1556] [Support]
I am facing `Unable to read key from file file:///tmp/storage/oauth-private.key` while deploying serverless 

I tried 
Run `php artisan passport:keys` locally to generate key files.
This command will generate the storage/oauth-private.key and storage/oauth-public.key files, which need to be deployed.

Depending on how you deploy your application (from your machine, or from CI), you may want to whitelist them in serverless.yml:

 ` package:
      patterns:
          - ...
          # Exclude the 'storage' directory
          - '!storage/**'
          # Except the public and private keys required by Laravel Passport
          - 'storage/oauth-private.key'
          - 'storage/oauth-public.key'`

`serverless deploy`

`  serverless bref:cli --args="passport:client --personal --name 'Laravel Personal Access Client'"
    serverless bref:cli --args="passport:client --password --name 'Laravel Personal Access Client'". `

#Laravel Framework 8.83.27
#serverless.yml
`service: app-serverless

provider:

    name: aws
    # The AWS region in which to deploy (us-east-1 is the default)
    region: me-south-1
    # Environment variables
    environment:
        APP_ENV: production # Or use ${sls:stage} if you want the environment to match the stage

package:

    # Files and directories to exclude from deployment
    patterns:
        - '!node_modules/**'
        - '!public/storage'
        - '!resources/assets/**'
        - '!storage/**'
        - '!tests/**'
        - 'storage/oauth-private.key'
        - 'storage/oauth-public.key'

functions:

    # This function runs the Laravel website/API
    web:

        handler: public/index.php
        runtime: php-81-fpm
        timeout: 28 # in seconds (API Gateway has a timeout of 29 seconds)
        events:
            - httpApi: '*'

    # This function lets us run artisan commands in Lambda
    artisan:

        handler: artisan
        runtime: php-81-console
        timeout: 720 # in seconds
        # Uncomment to also run the scheduler every minute
        #events:
        #    - schedule:
        #          rate: rate(1 minute)
        #          input: '"schedule:run"'

plugins:

    # We need to include the Bref plugin
    - ./vendor/bref/bref
`

`
#composer.json
  {
      "require": {
          "bref/bref": "^2.0",
          "bref/laravel-bridge": "^2.1",
      }
  }
`

Issue
<img width="1234" alt="Screenshot 2023-06-06 at 5 28 26 PM" src="https://github.com/brefphp/bref/assets/100999283/6de5113f-312b-4b65-baab-faee8e0d2eff">

**Comment by mnapoli** [comment]
Oh the new version of the Laravel Bridge moves the entire "storage" directory location to `/tmp`, so anything deployed in the app's `storage/` directory is ignored 🤔

I'm not sure if we can configure Laravel Passport to look at the keys in a specific folder?

**Reply by mnapoli** [reply]
cc @tillkruss / @georgeboot

**Reply by tillkruss** [reply]
I'm not familiar with Passport either. I think the .key is generated once and has to be placed inside the built Lambda artifact. I'm sure they offer an option to store it in a `/private` folder.

**Reply by georgeboot** [reply]
We just deployed the key and secret using ssm params and env variables.
